src/js/Controllers/TTSController.js

```diff
@@ -76,15 +105,63 @@ class TTSController {
                 } else if (this.filePlaylist[0].type === "TEXT"){
                     this.speak();
                 }    
+            } else {
+                this.speakEnded();
             }
         }
 
+        if (this.speechSynthesisInterval) {
+            clearInterval(this.speechSynthesisInterval);
+            this.speechSynthesisInterval = null;
+        }
+
+        this.currentlyPlaying = "TEXT";
         speechPlayer.text = text;
         speechPlayer.volume = 1;
         speechPlayer.rate = 1;
         speechPlayer.pitch = 0;
         window.speechSynthesis.speak(speechPlayer)
 
+        // Workaround for chrome issue where long utterances time out
+        this.speechSynthesisInterval = setInterval(() => {
+            if (!window.speechSynthesis.speaking) {
+                clearInterval(this.speechSynthesisInterval)
+                this.speechSynthesisInterval = null;
+            } else {
+                window.speechSynthesis.pause();
+                window.speechSynthesis.resume();
+            }
+        }, 14000);
+    }
+
+    stopSpeak(stopSpeakingID) {
+        const speakID = this.speakID;
+        this.speakID = null;
+        if (this.currentlyPlaying === "FILE") {
+            this.audioPlayer.onended = null;
+            this.audioPlayer.pause();
+            this.audioPlayer.src = "";
+        } else if (this.currentlyPlaying === "TEXT") {
+            window.speechSynthesis.pause();
+            window.speechSynthesis.cancel();
+        }
+        this.filePlaylist = [];
+        this.currentlyPlaying = null;
+        clearInterval(this.timers[this.speakID]);
```

Comment:
```suggestion
        clearInterval(this.timers[speakID]);
```
Ah, seems the previous suggestion was slightly off